### PR TITLE
Break github build script into four scripts, parallelize. 

### PR DESCRIPTION
**Pull request by leftwo**
## Break apart github buldomat steps

I took the draft work that @jclulow did and polished it.

break build-and-test.sh into four different steps, which
enables us to run three of them in parallel and enable
all the checks to complete in less time.

There is now just a build.sh which does the build and
cargo test.  At the end, it will copy binaries and some
selected scripts for future jobs to use.

test-up-encrypted.sh: just run test_up.sh encrypted

test-up-unencrypted.sh: just run test_up unencrypted

test-ds.sh: just run test_ds.sh

Updated test_up.sh and test_ds.sh to work better with buildomat

### CODE REVIEW

**Review by jmpesp — APPROVED** [review]
Cool!

File: .github/buildomat/jobs/build.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#:
+#: name = "build"
+#: variety = "basic"
+#: target = "helios"
+#: rust_toolchain = "nightly"
```
Comment (jclulow) [inline .github/buildomat/jobs/build.sh:6]: I think we should keep this in sync with `rust-toolchain.toml`, so in this case perhaps we should:

```
#: rust_toolchain = "nightly-2021-11-24"
```

**Review by jclulow — COMMENTED** [review]
This is great!  Thanks for carrying the drafts forward.  Make sure you squash out my XXX commit messages! :P

**Review by jclulow — APPROVED** [review]
LGTM!

**Pull request merged**